src/Microsoft.DotNet.Build.Tasks/PackageFiles/codeOptimization.targets

```diff
@@ -44,7 +82,7 @@
 
   <!-- We need the OptimizationData package in order to be able to optimize the assembly -->
   <Target Name="RestoreOptimizationDataPackage" BeforeTargets="CoreCompile"
-          Condition="'$(EnableProfileGuidedOptimization)'=='true' and !Exists('$(OptimizationDataDir)project.json')">
+          Condition="'$(EnableProfileGuidedOptimization)'=='true' and '$(RestoreDefaultOptimizationDataPackage)'=='true' and !Exists('$(OptimizationDataDir)project.json')">
```

Comment:
CoreCLR uses a different mechanism to restore the optdata package (I implemented it for PGO before IBC was up for consideration, and it's not a part of buildtools). Since there's no way to disable this task, by setting `$(OptimizationDataDir)` to enable IBC, I end up restoring additional unneeded data (since we never generated the project.json file in place here). It's messy, and rife with possibilities for collisions.

There's possibly a chance to unify and simplify the implementation (rewrite CoreCLR build a bit so that it can leverage this same restore for PGO and IBC data), but this is a smaller, safer, and I would argue also independently useful, change.